crud.py

```
@@ -0,0 +1,131 @@
+from datetime import date
+from typing import Sequence
+
+from fastapi import HTTPException
+from sqlalchemy import select
+from sqlalchemy.orm import Session
+
+from models import Author, Book
+from schemas import AuthorCreate, AuthorUpdate, BookCreate, BookUpdate
+from validation import validate_author_with_name_exists
+
+
+def get_all_authors(
+        db: Session,
+        skip: int = 0,
+        limit: int = 10
+) -> Sequence[Author]:
+    return db.scalars(select(Author).offset(skip).limit(limit)).all()
+
+
+def get_author_by_name(db: Session, name: str) -> Author:
+    return db.scalar(select(Author).where(Author.name == name))
+
+
+def get_author_by_id(db: Session, author_id: int) -> Author:
+    return db.scalar(select(Author).where(Author.id == author_id))
```

`get_author_by_id` returns `None` if the author is missing. Either raise `HTTPException(status_code=404)` here or update the return type to `Author | None` and let the route raise 404. Be consistent across your API so clients get a clear 404 when an item doesn't exist.